web/src/components/AssertionForm/AssertionForm.tsx

```diff
@@ -54,12 +55,21 @@ const AssertionForm: React.FC<TAssertionFormProps> = ({
   runId,
 }) => {
   const [form] = Form.useForm<IValues>();
-
   useGuidedTour(GuidedTours.Assertion);
 
   const currentSelectorList = Form.useWatch('selectorList', form) || [];
   const currentAssertionList = Form.useWatch('assertionList', form) || [];
-  const attributeList = useSelector(AssertionSelectors.selectAttributeList(testId, runId, currentSelectorList));
+  const currentPseudoSelector = Form.useWatch('pseudoSelector', form) || undefined;
+
+  const {data: spanIdList = []} = useGetSelectedSpansQuery({
+    query: SelectorService.getSelectorString(currentSelectorList, currentPseudoSelector),
+    testId,
+    runId,
+  });
+
+  const attributeList = useSelector<RootState, TSpanFlatAttribute[]>(state =>
```

Comment:
Please use the `useAppSelector` from `/redux/hooks.ts`. It already has the app state types.